src/Servers/Kestrel/Core/src/Internal/Http/ChunkWriter.cs

```diff
@@ -44,13 +44,14 @@ public static int BeginChunkBytes(int dataCount, Span<byte> span)
             return count;
         }
 
-        internal static void WriteBeginChunkBytes(this ref BufferWriter<PipeWriter> start, int dataCount)
+        internal static int WriteBeginChunkBytes(this ref BufferWriter<PipeWriter> start, int dataCount)
         {
             // 10 bytes is max length + \r\n
             start.Ensure(10);
```

Comment:
If the max length of the chunk prefix is actually 5 bytes, shouldn't we change this to Ensure(5)?

Reply:
I'd need to update the ChunkWriter tests, but as of now I believe you are correct.

Reply:
Actually I take that back, the max length can still be 10. If someone calls response.Body.WriteAsync(new byte[1000000]); this will still be written as one chunk. The chunk prefix restriction only applies to GetMemory/Advance.